frontend-plugin-core/src/main/java/com/github/eirslett/maven/plugins/frontend/lib/YarnInstaller.java

```diff
@@ -122,7 +123,23 @@ private void installYarn() throws InstallationException {
                 logger.warn("Failed to delete existing Yarn installation.");
             }
 
-            extractFile(archive, installDirectory);
+            try {
+                extractFile(archive, installDirectory);
+            } catch (ArchiveExtractionException e) {
+                if (e.getCause() instanceof EOFException) {
+                    // https://github.com/eirslett/frontend-maven-plugin/issues/794
+                    // The downloading was probably interrupted and archive file is incomplete:
+                    // delete it to retry from scratch
+                    this.logger.error("The archive file {} is corrupted and will be deleted. "
+                            + "Please try the build again.", archive.getPath());
+                    archive.delete();
+                    if (installDirectory.exists()) {
```

Comment:
The above code cleanups the installDirectory, but there is a probabilty to use corrupted (incomplete) Yarn dist.